modelci/finetuner/computer_vision_fine_tuning.py

```diff
@@ -152,203 +148,89 @@ def _unfreeze_and_add_param_group(
-
-        # 1. Load pre-trained network:
-        model_func = getattr(models, self.backbone)
-        backbone = model_func(pretrained=True)
-
-        _layers = list(backbone.children())[:-1]
-        self.feature_extractor = torch.nn.Sequential(*_layers)
-        freeze(module=self.feature_extractor, train_bn=self.train_bn)
-
-        # 2. Classifier:
-        _fc_layers = [torch.nn.Linear(2048, 256), torch.nn.Linear(256, 32), torch.nn.Linear(32, 1)]
-        self.fc = torch.nn.Sequential(*_fc_layers)
-
-        # 3. Loss:
-        self.loss_func = F.binary_cross_entropy_with_logits
+        self.save_hyperparameters('loss', 'batch_size', 'lr', 'lr_scheduler_gamma', 'step_size', 'num_workers')
 
     def forward(self, x):
-        """Forward pass. Returns logits."""
-
-        # 1. Feature extraction:
-        x = self.feature_extractor(x)
-        x = x.squeeze(-1).squeeze(-1)
-
-        # 2. Classifier (returns logits):
-        x = self.fc(x)
-
-        return x
-
-    def loss(self, logits, labels):
-        return self.loss_func(input=logits, target=labels)
-
-    def train(self, mode=True):
-        super().train(mode=mode)
-
-        epoch = self.current_epoch
-        if epoch < self.milestones[0] and mode:
-            # feature extractor is frozen (except for BatchNorm layers)
-            freeze(module=self.feature_extractor, train_bn=self.train_bn)
-
-        elif self.milestones[0] <= epoch < self.milestones[1] and mode:
-            # Unfreeze last two layers of the feature extractor
-            freeze(module=self.feature_extractor, n=-2, train_bn=self.train_bn)
-
-    def on_epoch_start(self):
-        """Use `on_epoch_start` to unfreeze layers progressively."""
-        optimizer = self.trainer.optimizers[0]
-        if self.current_epoch == self.milestones[0]:
-            _unfreeze_and_add_param_group(
-                module=self.feature_extractor[-2:], optimizer=optimizer, train_bn=self.train_bn
-            )
 
-        elif self.current_epoch == self.milestones[1]:
-            _unfreeze_and_add_param_group(
-                module=self.feature_extractor[:-2], optimizer=optimizer, train_bn=self.train_bn
-            )
+        return self.net(x)
 
     def training_step(self, batch, batch_idx):
 
         # 1. Forward pass:
         x, y = batch
-        y_logits = self.forward(x)
-        y_true = y.view((-1, 1)).type_as(x)
+        outputs = self.forward(x)
+        preds = torch.argmax(outputs, dim=1)
 
         # 2. Compute loss & accuracy:
-        train_loss = self.loss(y_logits, y_true)
-        accuracy = self.train_acc(y_logits, y_true)
+        train_loss = self.loss(outputs, y)
+        accuracy = self.train_acc(preds, y)
 
         # 3. Outputs:
-        tqdm_dict = {"train_loss": train_loss}
+        tqdm_dict = {'sloss': train_loss, 'train_acc': accuracy}
         self.log_dict(tqdm_dict, prog_bar=True)
-        return {"loss": train_loss}
+        return {"loss": train_loss, 'acc': accuracy}
```

Comment:
train_loss
train_acc

Reply:
`pytorch lightning` requires returning a dict that must include the key 'loss'